### ISSUE TITLE
Issue with deployment with 1.1 version

### ISSUE THREAD

**Issue by stefanSrecko** [pbi-tools/pbi-tools#351]
Hello,

Since the last update of pbi-tool we have encountered an issue with the deployment workflow.
Here are the details of `pbi-tools info`:
```
Run /app/pbi-tools/pbi-tools.core info
pbi-tools | Copyright (C) Mathias Thierbach 2018-2024

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU Affero General Public License as published
by the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

See <https://goto.pbi.tools/license> for full license details.


{
  "version": "1.1.0",
  "edition": "Core",
  "build": "1.1.24283.1830",
  "pbixProjVersion": "1.0",
  "amoVersion": "19.84.6.0",
  "toolPath": "/app/pbi-tools/pbi-tools.core",
  "locale": {
    "system": " (12[7](https://github.com/xxxx/analytics-power-bi/actions/runs/11288244767/job/31395634437#step:4:8))",
    "ui": " (127)"
  },
  "settings": {
    "PBITOOLS_LogLevel": null,
    "PBITOOLS_PbiInstallDir": null,
    "PBITOOLS_AppDataDir": null,
    "PBITOOLS_Debug": null,
    "PBITOOLS_UICulture": null,
    "PBITOOLS_EffectiveDate": null,
    "PBITOOLS_DefaultModelSerialization": null
  },
  "runtime": {
    "platform": "Debian GNU/Linux 12 (bookworm)",
    "architecture": "X64",
    "framework": ".NET [8](https://github.com/xxxx/analytics-power-bi/actions/runs/11288244767/job/31395634437#step:4:9).0.10"
  }
}
```

And here is the message we recieve on the deployment task:
```
Run /app/pbi-tools/pbi-tools.core deploy "./Datasets/Hourly Reporting" "$PROFILE" Test
pbi-tools | Copyright (C) Mathias Thierbach 201[8](https://github.com/thrasio/analytics-power-bi/actions/runs/11288244767/job/31395634437#step:6:9)-2024

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU Affero General Public License as published
by the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

See <https://goto.pbi.tools/license> for full license details.


Reading PBIXPROJ settings from: /__w/analytics-power-bi/analytics-power-bi/Datasets/Hourly Reporting/.pbixproj.json
Starting deployment 'Hourly Reporting' into environment: Test ...
Reading PBIXPROJ settings from: /__w/analytics-power-bi/analytics-power-bi/Datasets/Hourly Reporting/.pbixproj.json
Access token received. Expires On: [10](https://github.com/thrasio/analytics-power-bi/actions/runs/11288244767/job/31395634437#step:6:11)/11/2024 08:32:46 +00:00
Resolved workspace ID '18e56ef6-e0c2-4b63-847c-xxxxxxxxx' for workspace: 'Corporate Datasets [Test]'
Connecting to XMLA endpoint: powerbi://api.powerbi.com/v1.0/myorg/Corporate Datasets [Test]
An unhandled exception occurred.
System.AggregateException: One or more errors occurred. (The type initializer for 'Microsoft.AnalysisServices.Hosting.ClientHostingManager' threw an exception.)
 ---> System.TypeInitializationException: The type initializer for 'Microsoft.AnalysisServices.Hosting.ClientHostingManager' threw an exception.
 ---> System.ArgumentNullException: Value cannot be null. (Parameter 'path1')
   at System.ArgumentNullException.Throw(String paramName)
   at System.IO.Path.Combine(String path1, String path2)
   at Microsoft.AnalysisServices.Configuration.ClientConfigLoader.AddDefaultConfigurationValues(IDictionary`2 config)
   at Microsoft.AnalysisServices.Configuration.ClientConfigLoader.LoadClientConfiguration()
   at Microsoft.AnalysisServices.Hosting.ClientHostingManager.InitializeRuntimeAndClientConfiguration()
   at Microsoft.AnalysisServices.Hosting.ClientHostingManager.InitializeConfiguration()
   at Microsoft.AnalysisServices.Hosting.ClientHostingManager.Initialize()
   at Microsoft.AnalysisServices.Hosting.ClientHostingManager..cctor()
   --- End of inner exception stack trace ---
   at Microsoft.AnalysisServices.Hosting.ClientHostingManager.get_IsProcessWithUserInterface()
   at Microsoft.AnalysisServices.ConnectionInfo.HandlePaaSInfrastructureConnection(IConnectivityOwner owner)
   at Microsoft.AnalysisServices.ConnectionInfo.ValidateAndCompleteConnectionConfiguration(IConnectivityOwner owner)
   at Microsoft.AnalysisServices.ConnectionInfo.SetConnectionString(String cs)
   at Microsoft.AnalysisServices.ConnectionInfo..ctor(String connectionString)
   at Microsoft.AnalysisServices.XmlaClient.Connect(String connectionString, Boolean beginSession)
   at Microsoft.AnalysisServices.Core.Server.Connect(String connectionString, String sessionId, ObjectExpansion expansionType)
   at Microsoft.AnalysisServices.Core.Server.Connect(String connectionString, String sessionId)
   at Microsoft.AnalysisServices.Core.Server.Connect(String connectionString)
   at PbiTools.Deployments.DeploymentManager.DeployDatasetAsync(PbiDeploymentManifest manifest, String label, String environment) in X:\pbi-tools\BRANCHES\1.1\src\PBI-Tools\Deployments\DeploymentManager.Dataset.cs:line [13](https://github.com/thrasio/analytics-power-bi/actions/runs/11288244767/job/31395634437#step:6:14)1
   at PbiTools.Deployments.DeploymentManager.DeployAsync(String profileName, String environment) in X:\pbi-tools\BRANCHES\1.1\src\PBI-Tools\Deployments\DeploymentManager.cs:line 106
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at PbiTools.Cli.CmdLineActions.Deploy(String folder, String label, String environment, String basePath, Boolean whatIf) in X:\pbi-tools\BRANCHES\1.1\src\PBI-Tools\Cli\Deploy.cs:line 56
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithManyArgs(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
--- End of stack trace from previous location ---
   at PowerArgs.ArgAction.Invoke()
   at PbiTools.Program.Main(String[] args) in X:\pbi-tools\BRANCHES\1.1\src\PBI-Tools\Program.cs:line 133
Error: Process completed with exit code [24](https://github.com/xxxx/analytics-power-bi/actions/runs/11288244767/job/31395634437#step:6:25)7.
```

Interestingly, by updating deployment workflow yml file from:
```
jobs:
  Deployment:
    runs-on: ubuntu-latest
    environment: Test
    container:
      image: ghcr.io/pbi-tools/pbi-tools-core:latest
```
to
```
jobs:
  Deployment:
    runs-on: ubuntu-latest
    environment: Test
    container:
      image: ghcr.io/pbi-tools/pbi-tools-core:1.0.1
```
made our deployment succeed.

We would appreciate any help on how can we overcome this error and continue using the latest app image.
Thank you!

**Comment by mthierba** [comment]
Thanks for reporting. I'll review shortly. For now you can revert to 1.0.1 as you have done already.

**Comment by mthierba** [comment]
This has been resolved in the new 1.1.1 release. Microsoft's TOM library apparently introduced a bug that is specific to Linux and effectively prohibits all deployments. I was able to implement a workaround (and made the product team aware).
This release also adds automated integration tests to prevent future regressions. See the latest run here: https://github.com/pbi-tools/pbi-tools/actions/runs/11322724054
